lib/internal/bootstrap/node.js

```diff
@@ -311,6 +292,43 @@
     }
   }
 
+  function setupRepl() {
+    // used by https://github.com/nodejs/repl
+    if (process.env.NODE_EXTERNAL_REPL_MODULE) {
+      const { makeRequireFunction } =
+        NativeModule.require('internal/modules/cjs/helpers');
+      const CJSModule = NativeModule
+        .require('internal/modules/cjs/loader');
+      const replRoot = new CJSModule('<repl root>');
+      replRoot.paths = CJSModule
```

Comment:
The default repl has no parent module. Is one, `<repl root>` strictly needed here?

Reply:
@jdalton yes. NativeModule has no concept of parents, but it can't deal with files. we have to construct an actual requirejs root here.

Reply:
@devsnek Construct it along how  a [main module](https://github.com/nodejs/node/blob/c3f8dd6f98328a9a47814f1856bc0914859c6e19/lib/internal/modules/cjs/loader.js#L742) is handled but with an `undefined` instead of `null` parent.